### PR TITLE
CLN: drop **kwds from pd.read_excel

### PR DESCRIPTION
**Pull request by topper-123**
Drop ``**kwds`` from ``pd.read_excel``.

Wrt. "new" keyword parameter ``na_filter``: This parameter already exists in the doc string and is just masked in the ``kwds`` in the signature. If you follow the code paths all the way down to parsers.py, you can see that it indeed has a default of `True``.

### CODE REVIEW

**Comment by WillAyd** [comment]
Lgtm if green

File: pandas/tests/io/excel/test_readers.py [hunk 1]
```diff
@@ -897,12 +897,6 @@ def test_read_excel_bool_header_arg(self, read_ext):
             with pytest.raises(TypeError, match=msg):
                 pd.read_excel("test1" + read_ext, header=arg)
 
-    def test_read_excel_chunksize(self, read_ext):
```
Comment (jreback) [inline pandas/tests/io/excel/test_readers.py:900]: does this work now?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/excel/test_readers.py (hunk 1 above)
Comment (topper-123) [reply pandas/tests/io/excel/test_readers.py:900]: I've removed ``**kwds`` from the signature, so passing ``chunksize`` into ``read_excel`` isn't even possible any more and testing if it fails makes no sense now.

More generally, previously any keyword argument could be passed, so e.g. ``read_excel("file.xlsx", blabla=1)`` didn't fail, but shoild fail.

Comment (jreback) [reply pandas/tests/io/excel/test_readers.py:900]: ok can you add a test then for extraneous keywords?

Comment (topper-123) [reply pandas/tests/io/excel/test_readers.py:900]: IMO that would be too preemptive. Someone could of course later add a ``**kwargs`` but they could also do so much else wrong...

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**